### PR TITLE
chore: move sdk loader to separate file

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## PR Description

- Create a separate file for loading SDKs

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/Axeptio/loader.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { LOAD_ORIGIN } from '../../utils/ScriptLoader';
+
+/* eslint-disable */
+// START-NO-SONAR-SCAN
```
Comment (saikumarrs) [inline src/integrations/Axeptio/loader.js:4]: We need to rename these files to `nativeSdkLoader.js`.
Then, these guarding comments to disable scans should be removed.
See this PR for reference, https://github.com/rudderlabs/rudder-sdk-js/commit/caa2d34fbf4f6619c2cfdc09b267b928d5a1c7f9.

File: src/integrations/Axeptio/loader.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+function loader(d, s, clientId) {
```
Comment (saikumarrs) [inline src/integrations/Axeptio/loader.js:5]: `loadNativeSdk` is a better name. If you change the function names, ensure to rename it in Braze also for consistency.

File: src/integrations/LiveChat/browser.js [hunk 3]
```diff
@@ -29,40 +30,8 @@ class LiveChat {
     logger.debug('===in init Livechat===');
     window.__lc = window.__lc || {};
     window.__lc.license = this.licenseId;
```
Comment (saikumarrs) [inline src/integrations/LiveChat/browser.js:32]: Isn't this also part of the loader logic?

File: src/integrations/Matomo/browser.js [hunk 4]
```diff
@@ -55,19 +55,7 @@ class Matomo {
 
   loadScript() {
     window._paq = window._paq || [];
```
Comment (saikumarrs) [inline src/integrations/Matomo/browser.js:57]: This should also be part of the loader script.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

**Review by koladilip — DISMISSED** [review]

File: src/integrations/Axeptio/loader.js (hunk 2 above)
Comment (ujjwal-ab) [reply src/integrations/Axeptio/loader.js]: Updated

File: src/integrations/LiveChat/browser.js (hunk 3 above)
Comment (ujjwal-ab) [reply src/integrations/LiveChat/browser.js]: Updated

File: src/integrations/Matomo/browser.js (hunk 4 above)
Comment (ujjwal-ab) [reply src/integrations/Matomo/browser.js]: Updated

**Review by saikumarrs — COMMENTED** [review]
I just realized I've not published these comments. Apologies.

File: src/integrations/BingAds/nativeSdkLoader.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    n.setAttribute('data-loader', LOAD_ORIGIN),
+    (n.onload = n.onreadystatechange =
+      function () {
+        const s = this.readyState;
+        (s && s !== 'loaded' && s !== 'complete' && typeof w['UET'] === 'function') ||
+          (f(), (n.onload = n.onreadystatechange = null));
+      }),
+    (i = d.getElementsByTagName(t)[0]),
+    i.parentNode.insertBefore(n, i);
+}
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/integrations/BingAds/nativeSdkLoader.js]: Updated

**Review by saikumarrs — APPROVED** [review]

File: src/integrations/Comscore/nativeSdkLoader.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { LOAD_ORIGIN } from '../../utils/ScriptLoader';
+
+function loadNativeSdk() {
+  const s = document.createElement('script');
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/integrations/Comscore/nativeSdkLoader.js]: Done

File: src/integrations/Axeptio/browser.js [hunk 7]
```diff
@@ -21,22 +21,9 @@ class Axeptio {
...
-      e.async = true;
-      e.src = '//static.axept.io/sdk.js';
-      e.setAttribute('data-loader', LOAD_ORIGIN), t.parentNode.insertBefore(e, t);
-    })(document, 'script');
-  }
-
   init() {
     logger.debug('===In init Axeptio===');
-    this.loadScript();
+    loadNativeSdk(document, 'script', this.clientId);
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/integrations/Axeptio/browser.js]: Updated

File: src/integrations/BingAds/browser.js [hunk 8]
```diff
@@ -18,37 +18,8 @@ class BingAds {
...
-        (i = d.getElementsByTagName(t)[0]),
-        i.parentNode.insertBefore(n, i);
-    })(window, document, 'script', 'https://bat.bing.com/bat.js', this.uniqueId);
-  };
-  /* eslint-enable */
-
   init = () => {
-    this.loadBingadsScript();
+    loadNativeSdk(this.uniqueId);
     logger.debug('===in init BingAds===');
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply src/integrations/BingAds/browser.js:22]: Updated

File: src/integrations/Comscore/nativeSdkLoader.js [hunk 9]
```diff
@@ -1,16 +1,11 @@
...
-    s.src = `${
-      document.location.protocol == 'https:' ? 'https://sb' : 'http://b'
-    }.scorecardresearch.com/beacon.js`;
-    el.parentNode.insertBefore(s, el);
-  })();
+  if (document.location.protocol == 'https:') {
+    ScriptLoader('comscore', 'https://sb.scorecardresearch.com/beacon.js');
+  } else {
+    ScriptLoader('comscore', 'http://b.scorecardresearch.com/beacon.js');
+  }
```
Comment (saikumarrs) [inline src/integrations/Comscore/nativeSdkLoader.js:8]:
```suggestion
  const src = `${
      document.location.protocol == 'https:' ? 'https://sb' : 'http://b'
    }.scorecardresearch.com/beacon.js`;

  ScriptLoader('comscore', src);
```

**Review by saikumarrs — APPROVED** [review]

File: src/integrations/Comscore/nativeSdkLoader.js (hunk 9 above)
Comment (ujjwal-ab) [reply src/integrations/Comscore/nativeSdkLoader.js]: updated

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**